esp/esdllib/esdl_def.cpp

```diff
@@ -1288,22 +1288,10 @@ class EsdlDefServiceArrayIterator : public CInterface, implements IEsdlDefServic
     {
     }
 
-    bool        first(void)             { cur = 0; return isValid(); }
-    bool        isValid(void)           { return array.isItem(cur); }
-    IEsdlDefService &query()                { assertex(isValid()); return array.item(cur); }
-    bool        hasNext(void)           { return array.isItem(cur+1); }
-    bool        hasPrev(void)           { return array.isItem(cur-1); }
-    bool        last(void)              { cur = array.ordinality()-1; return isValid(); }
-    bool        next(void)              { ++cur; return isValid(); }
-    bool        prev(void)              { --cur; return isValid(); }
-    bool        select(aindex_t seek)   { cur = seek; return isValid(); }
-
-
-    unsigned getFlags() { return 0; }
-    IEsdlDefObjectIterator* queryBaseTypesIterator()
-    {
-        return NULL;
-    }
+    virtual bool        first(void)             override { cur = 0; return isValid(); }
+    virtual bool        isValid(void)           override { return array.isItem(cur); }
+    virtual IEsdlDefService &query()                override { assertex(isValid()); return array.item(cur); }
+    virtual bool        next(void)              override { ++cur; return isValid(); }
```

Comment:
Whitespace... I personally would not put so much space between the method signature and the implementation, but with so much space present there is no reason for the implementation of `query` to be indented even more.